src/registries/npm-registry.js

```diff
@@ -98,7 +98,7 @@ export default class NpmRegistry extends Registry {
   async getPossibleConfigLocations(filename: string): Promise<Array<[boolean, string, string]>> {
     const possibles = [
       [false, path.join(this.cwd, filename)],
-      [true, this.config.userconfig || path.join(userHome, filename)],
+      [true, this.config.userconfig || path.join(process.env.YARN_HOME_FOLDER || userHome, filename)],
```

Comment:
This is a bit confusing because we have HOME defined in constants.js, npm-registry and yarn-registry.
Could you bring all of them into one place?